src/client.ts

```diff
@@ -124,22 +111,22 @@ export class APIClient {
     return devices;
   }
 
-  //retrieves any user group ids assigned to this application
+  //retrieves any user groups assigned to this application
   public async getGroupsForApp(appId) {
-    const groupIds: string[] = [];
+    const groups: OktaApplicationGroup[] = [];
     await this.oktaClient.listApplicationGroupAssignments(appId).each((e) => {
-      groupIds.push(e.id);
+      groups.push(e);
     });
-    return groupIds;
+    return groups;
```

Comment:
Just crossed my mind, but it looks like we could change this to `iterateGroupsForApp(appId, iteratee)` so that we don't have to load all the data in memory (we can "stream process" the groups).